h2/src/main/org/h2/mvstore/db/MVPrimaryIndex.java

```diff
@@ -72,7 +72,7 @@ public MVPrimaryIndex(Database db, MVTable table, int id,
         if (!table.isPersistData()) {
             dataMap.map.setVolatile(true);
         }
-        Value k = dataMap.lastKey();
+        Value k = dataMap.map.lastKey();    // include uncommitted keys as well
```

Comment:
but we've committed the map a couple of lines above?

Reply:
Yes, we've committed a transaction which opens the map. BTW, I think API is a little backward here and it should be possible to create TransactionMap just from MVMap and Transaction, and as part of the index store just reference to MVMap, but anyway, what if there is another transaction which inserted rows into the table, but haven't committed before store was closed/crashed. It is still pending at this point.